### ISSUE TITLE
Regression: wide lines in code cause posts to spill under right column, making questions unreadable

### ISSUE THREAD

**Issue by cellio** [codidact/qpixel#169]
Here are two questions that demonstrate the problem:

https://software.codidact.com/questions/277428

Interestingly, some code blocks go over and some under.  Notice the usercard is buried.

![Screen Shot 2020-09-02 at 10 49 47 PM](https://user-images.githubusercontent.com/5557942/92065959-b3fb5580-ed6e-11ea-89aa-f8cfe5e9f1ec.png)

https://software.codidact.com/questions/277881

![Screen Shot 2020-09-02 at 10 51 22 PM](https://user-images.githubusercontent.com/5557942/92066007-da20f580-ed6e-11ea-8af3-cf326e5559d5.png)

And here's one that had the problem, but I edited the code block to hack around it so the page would display: https://software.codidact.com/questions/277885

I'm saying this is a regression because I've seen that first question before, and when it was first posted it did not have this problem.  Here is a copy from Internet Archive demonstrating that: https://web.archive.org/web/20200822170202/https://software.codidact.com/questions/277428 (h/t Moshi). 
That question was posted on 2020-08-22, so sometime between then and now this changed.  We've had a couple updates in that window, so I'm hoping that someone who knows what to look for can look at recent commits and spot it.

**Comment by cellio** [comment]
Presumably related meta post: https://software.codidact.com/questions/277551

**Comment by MoshiKoi** [comment]
The specific change comes from here:

In the archived (functional) version, the post looks like this
```html
<div class="post--container  grid is-nowrap">
    <div class="grid--cell">...</div>
    <div class="grid--cell is-flexible has-padding-1">...</div>
</div>
```

In the current (non-functional) version, it looks like this
```html
<div class="post--container  grid is-nowrap">
    <div>...</div>
    <div class="has-padding-2">...</div>
</div>
```

Notably, the "grid--cell" and "is-flexible" classes are missing in the newer version. Adding them back in with Inspect Element fixes the issue.

**Comment by galegosimpatico** [comment]
Reproduced locally using both 26df584 (same as production) and `develop` current tip 489117e.

> Interestingly, some code blocks go over and some under. Notice the usercard is buried.

I think all which have vertical scroll go over while all which have not go under.

**Comment by galegosimpatico** [comment]
Bisected to 9100edf7d2a1.

**Comment by galegosimpatico** [comment]
I'd just wipe out this line and redeploy to production:https://github.com/codidact/qpixel/blob/9100edf7d2a1116dce380ae496de9e2dd1493750/app/assets/stylesheets/posts.scss#L126

That's gonna enforce mobile layout for desktop clients, but salvaging readability. I think it's a reasonable emergency hotfix while you figure out a definitive fix.

<img width="1053" alt="Screenshot 2020-09-03 at 14 40 11" src="https://user-images.githubusercontent.com/53513496/92115887-6543cf00-edf3-11ea-94f2-82058adfb73f.png">

cheers!

**Comment by cellio** [comment]
Adding the following CSS, provided by Moshi, to Stylus fixed the overflow for me:

.post--container > div:nth-child(2) {
    width: 97%;
}

At 100% the vertical scrollbars overflowed and text got clipped on the right -- presumably it was aligned with the outer edge of those scrollbars.  Dropping it down a few % fixed that.  Here it is at 100%:

![Screen Shot 2020-09-03 at 9 18 38 PM](https://user-images.githubusercontent.com/5557942/92188875-c091b380-ee2b-11ea-8987-cc9f8bd93256.png)

**Comment by luap42** [comment]
Yeah. I’ll look into it later today/tomorrow. However, setting the width to 97% is likely not the correct answer, as it will probably don’t work as responsive, because the margins aren’t in % but in px.

**Comment by cellio** [comment]
Oh, I thought the % was scoped to its container.  My 97% isn't using 97% of the whole viewport; the right column is intact.  But anyway, I'm not trying to say how we should implement it, just reporting what allowed me to work around the problem.

**Comment by MoshiKoi** [comment]
@luap42 can't we just restore the classes that were deleted, since that fixes the problem?

**Comment by ArtOfCode-** [comment]
Not necessarily, @MoshiKoi, because there have been CSS changes as well that rely on some of those classes _not_ being there.

**Comment by cellio** [comment]
Did we fix this some other way?  I don't see a linked PR but the issue is closed.

**Comment by ArtOfCode-** [comment]
Fixed in bfa7b493cd2e28e61005882a12e2699953977466
